src/server/nfc/index.test.ts

```diff
@@ -0,0 +1,479 @@
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't add attendee");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		afterAll(async () => {
+			try {
+				const ret = await models.Events.findOneAndUpdate(
+					{ _id: testEventId },
+					{
+						$push: {
+							attendees: [],
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't add attendee");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		it('check event should warn', async () => {
+			await expect(
+				shouldWarnRepeatedCheckIn(testHackerId.toHexString(), testEventId.toHexString(), models)
+			).resolves.toEqual(true);
+		});
+		it('check user not in event', async () => {
+			await expect(
+				userIsAttendingEvent(testHackerId2.toHexString(), testEventId.toHexString(), models)
+			).resolves.toEqual(false);
+		});
+	});
+	describe('getEventsAttended', () => {
+		const testEventsArray = ['TEST_EVENT_1', 'TEST_EVENT_2'];
+		beforeAll(async () => {
+			try {
+				const ret = await models.Hackers.findOneAndUpdate(
+					{ _id: testHackerId },
+					{
+						$set: {
+							eventsAttended: testEventsArray,
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't set events attended");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		afterAll(async () => {
+			try {
+				const ret = await models.Hackers.findOneAndUpdate(
+					{ _id: testHackerId },
+					{
+						$set: {
+							eventsAttended: [],
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't set events attended");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		it('get events attended', async () => {
+			await expect(getEventsAttended(testHackerId.toHexString(), models)).resolves.toEqual(
+				testEventsArray
+			);
+		});
+	});
+	describe('getAttendees', () => {
+		const testAttendeesArray = ['TEST_ATTENDEE_1', 'TEST_ATTENDEE_2'];
+		beforeAll(async () => {
+			try {
+				const ret = await models.Events.findOneAndUpdate(
+					{ _id: testEventId },
+					{
+						$set: {
+							attendees: testAttendeesArray,
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't set attendees");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		afterAll(async () => {
+			try {
+				const ret = await models.Events.findOneAndUpdate(
+					{ _id: testEventId },
+					{
+						$set: {
+							attendees: [],
+						},
+					}
+				);
+				if (!ret.value) throw new MongoError("Can't set attendees");
+			} catch (err) {
+				// eslint-disable-next-line no-console
+				console.error(err);
+			}
+		});
+		it('get events attended', async () => {
+			await expect(getAttendees(testEventId.toHexString(), models)).resolves.toEqual(
+				testAttendeesArray
+			);
+		});
+	});
+});
```

Comment:
Let's add two integration tests:
1. adding attendee then removing.
2. adding attendee then checking attendance.

Reply:
An integration test was added that adds attendees, checks attendance, and then removes attendees